### PR TITLE
rustfmt tests

### PR DESCRIPTION
**Pull request by matthiaskrgr**
rustfmt tests and update stderr.
I added a couple `#[rustfmt::skip]`s

### CODE REVIEW

**Comment by matthiaskrgr** [comment]
Should we ensure that tests are formatted via `ci/base-tests.sh`?
Unfortunately this will be a bit more tricky than running `cargo fmt` since there is not "tests-crate" but I guess we could loop over all files and fmt --check every one of them...

File: tests/ui/empty_line_after_outer_attribute.rs [hunk 1]
```diff
@@ -7,66 +7,70 @@
...
-fn with_one_newline() { assert!(true) }
+fn with_one_newline() {
+    assert!(true)
+}
 
 // This should produce a warning, too
 #[crate_type = "lib"]
 
-
-fn with_two_newlines() { assert!(true) }
```
Comment (phansch) [inline tests/ui/empty_line_after_outer_attribute.rs]: this needs to stay as before, as the lint is 'parsing' the span between the attribute and the following item. This test makes sure newlines are handled correctly.

File: tests/ui/doc.rs [hunk 2]
```diff
@@ -167,9 +151,9 @@ fn issue_902_comment() {}
 fn issue_1469() {}
 
 /**
- * This is a doc comment that should not be a list
- *This would also be an error under a strict common mark interpretation
- */
+ ** This is a doc comment that should not be a list
```
Comment (oli-obk) [inline tests/ui/doc.rs]: uuh, I have no idea what this change does

File: tests/ui/double_parens.rs [hunk 3]
```diff
@@ -23,7 +19,7 @@ impl DummyStruct {
 }
 
 fn simple_double_parens() -> i32 {
-    ((0))
```
Comment (oli-obk) [inline tests/ui/double_parens.rs]: missing rustfmt_skip... or we just remove the lint entirely as it's covered by rustfmt?

File: tests/ui/eq_op.stderr [hunk 4]
```diff
@@ -37,131 +37,131 @@ error: this boolean expression can be simplified
...
    |
-28 |     (1 as u64) & (1 as u64);
+29 |     (1 as u64) & (1 as u64);
    |     ^^^^^^^^^^^^^^^^^^^^^^^
 
 error: equal expressions as operands to `^`
-  --> $DIR/eq_op.rs:29:5
+  --> $DIR/eq_op.rs:30:5
    |
-29 |     1 ^ ((((((1))))));
```
Comment (oli-obk) [inline tests/ui/eq_op.stderr]: rustfmt_skip missing

File: tests/ui/unused_unit.stderr [hunk 5]
```diff
@@ -1,15 +1,13 @@
 error: unneeded unit return type
-  --> $DIR/unused_unit.rs:28:59
+  --> $DIR/unused_unit.rs:27:59
    |
-28 |       pub fn get_unit<F: Fn() -> (), G>(&self, f: F, _g: G) ->
-   |  ___________________________________________________________^
-29 | |         ()
-   | |__________^ help: remove the `-> ()`
+27 |     pub fn get_unit<F: Fn() -> (), G>(&self, f: F, _g: G) -> ()
```
Comment (oli-obk) [inline tests/ui/unused_unit.stderr]: I think this was specifically testing newlines

**Pull request closed without merging**